Templates/DefaultGem/Template/Code/CMakeLists.txt

```diff
@@ -156,7 +156,9 @@ endif()
 # See if globally, tests are supported
 if(PAL_TRAIT_BUILD_TESTS_SUPPORTED)
     # We globally support tests, see if we support tests on this platform for ${Name}.Tests
-    if(PAL_TRAIT_${NameUpper}_TEST_SUPPORTED)
+    # Remove this line if Tests are added, and uncomment the second line
+    if(0)
+    # if(PAL_TRAIT_${NameUpper}_TEST_SUPPORTED)
```

Comment:
Instead of commenting out these lines for disabling testing for a Gem, we can instead update the 
default value of `PAL_TRAIT_${NameUpper}_TEST_SUPPORTED` to `FALSE` in the [PAL_Windows.cmake](https://github.com/o3de/o3de/blob/development/Templates/DefaultGem/Template/Code/Platform/Windows/PAL_windows.cmake), [PAL_iOS.cmake](https://github.com/o3de/o3de/blob/development/Templates/DefaultGem/Template/Code/Platform/iOS/PAL_ios.cmake), [PAL_Linux.cmake](https://github.com/o3de/o3de/blob/development/Templates/DefaultGem/Template/Code/Platform/Linux/PAL_linux.cmake), [PAL_Android.cmake](https://github.com/o3de/o3de/blob/development/Templates/DefaultGem/Template/Code/Platform/Android/PAL_android.cmake) and [PAL_Mac.cmake](https://github.com/o3de/o3de/blob/development/Templates/DefaultGem/Template/Code/Platform/Mac/PAL_mac.cmake) files in each gem template.